## SVM Quant materials

https://www.quantstart.com/articles/Support-Vector-Machines-A-Guide-for-Beginners/

### Importing Clean Dataset

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('cleaned_filghts.csv')

In [3]:
df.tail(50)

Unnamed: 0  YEAR  MONTH  DAY  DAY_OF_WEEK AIRLINE  FLIGHT_NUMBER  \
5332864     5332864  2015     12    3            4      EV           2730   
5332865     5332865  2015     12    3            4      EV           2734   
5332866     5332866  2015     12    4            5      EV           2730   
5332867     5332867  2015     12    4            5      EV           2734   
5332868     5332868  2015     12    5            6      EV           2734   
5332869     5332869  2015     12    6            7      EV           2734   
5332870     5332870  2015     12    7            1      EV           2730   
5332871     5332871  2015     12    7            1      EV           2734   
5332872     5332872  2015     12    8            2      EV           2730   
5332873     5332873  2015     12    8            2      EV           2734   
5332874     5332874  2015     12    9            3      EV           2730   
5332875     5332875  2015     12    9            3      EV           2734   
5332876     5332876  2015     12   10            4      EV           2730   
5332877     5332877  2015     12   10            4      EV           2734   
5332878     5332878  2015     12   11            5      EV           2730   
5332879     5332879  2015     12   11            5      EV           2734   
5332880     5332880  2015     12   12            6      EV           2734   
5332881     5332881  2015     12   13            7      EV           2734   
5332882     5332882  2015     12   14            1      EV           2730   
5332883     5332883  2015     12   14            1      EV           2734   
5332884     5332884  2015     12   15            2      EV           2730   
5332885     5332885  2015     12   15            2      EV           2734   
5332886     5332886  2015     12   16            3      EV           2730   
5332887     5332887  2015     12   16            3      EV           2734   
5332888     5332888  2015     12   17            4      EV           2730   
5332889     5332889  2015     12   17            4      EV           2734   
5332890     5332890  2015     12   18            5      EV           2730   
5332891     5332891  2015     12   18            5      EV           2734   
5332892     5332892  2015     12   19            6      EV           2734   
5332893     5332893  2015     12   20            7      EV           2734   
5332894     5332894  2015     12   21            1      EV           2730   
5332895     5332895  2015     12   21            1      EV           2734   
5332896     5332896  2015     12   22            2      EV           2730   
5332897     5332897  2015     12   22            2      EV           2734   
5332898     5332898  2015     12   23            3      EV           2730   
5332899     5332899  2015     12   23            3      EV           2734   
5332900     5332900  2015     12   24            4      EV           2730   
5332901     5332901  2015     12   24            4      EV           2734   
5332902     5332902  2015     12   25            5      EV           2730   
5332903     5332903  2015     12   25            5      EV           2734   
5332904     5332904  2015     12   26            6      EV           2734   
5332905     5332905  2015     12   27            7      EV           2734   
5332906     5332906  2015     12   28            1      EV           2730   
5332907     5332907  2015     12   28            1      EV           2734   
5332908     5332908  2015     12   29            2      EV           2730   
5332909     5332909  2015     12   29            2      EV           2734   
5332910     5332910  2015     12   30            3      EV           2730   
5332911     5332911  2015     12   30            3      EV           2734   
5332912     5332912  2015     12   31            4      EV           2730   
5332913     5332913  2015     12   31            4      EV           2734   

        TAIL_NUMBER ORIGIN_AIRPORT DESTINATION_AIRPORT  ...  WEATHER_DELAY  \
533

In [19]:
state_totals = df.groupby(['ORIGIN_STATE']).ORIGIN_STATE.agg('count')

In [20]:
state_totals

ORIGIN_STATE
AK     33300
AL     26482
AR     22654
AS       107
AZ    165469
CA    647911
CO    215451
CT     18799
DE       100
FL    415586
GA    360496
GU       334
HI     91181
IA     17382
ID     16915
IL    381644
IN     39107
KS     10966
KY     36880
LA     65185
MA    109290
MD     86079
ME      5047
MI    137033
MN    118701
MO     93838
MS     13109
MT     14848
NC    146906
ND     16530
NE     20023
NH      6292
NJ    108174
NM     22241
NV    146723
NY    246235
OH     74514
OK     31445
OR     55620
PA     99995
PR     26748
RI     11058
SC     28341
SD      9267
TN     75443
TX    631124
UT     99972
VA    138577
VI      5287
VT      2978
WA    123548
WI     51376
WV      2385
WY      8218
Name: ORIGIN_STATE, dtype: int64

In [17]:
airline_totals = df.groupby(['AIRLINE']).AIRLINE.agg('count')

In [29]:
airline_totals

AIRLINE
AA     648694
AS     158054
B6     245135
DL     800329
EV     526249
F9      82735
HA      70030
MQ     272650
NK     107171
OO     539545
UA     469829
US     198715
VX      56439
WN    1157339
Name: AIRLINE, dtype: int64

### Let's select Texas as a large state => 631,124 flights in 2015

### And Wisconsin as a 50k pa state to test the workflow!

In [21]:
wisconsin = df[df['ORIGIN_STATE'] == 'WI']

In [24]:
wisconsin.to_csv('wisconsin.csv')

In [26]:
big_texas = df[df['ORIGIN_STATE'] == 'TX']

In [28]:
big_texas.to_csv('big_texas.csv')